fe/fe-core/src/main/java/org/apache/doris/job/task/AbstractTask.java

```diff
@@ -184,7 +184,9 @@ public void runTask() throws JobException {
             onFail();
             log.warn("execute task error, job id is {}, task id is {}", jobId, taskId, e);
         } finally {
-            closeOrReleaseResources();
+            if (!TaskStatus.CANCELED.equals(status)) {
+                closeOrReleaseResources();
```

Comment:
why only release resource when status is not canceled? add some comment to explain it

Reply:
The cancel logic will call the closeOrReleased Resources method by itself. If it is also called here, it may result in the inability to obtain relevant information when canceling the task

Reply:
Mtmv is initialized in the before method. If the task has not yet run, the before method will not be used, so mtmv will be empty, which prevents the canceled task from being added to the history list